src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1626,13 +1626,13 @@ public static IEnumerable<IEdmStructuralProperty> Key(this IEdmEntityType type)
         /// <summary>
         /// Adds the alternate keys to this entity type.
         /// </summary>
-        /// <param name="type">Reference to the calling object.</param>
         /// <param name="model">The model to be used.</param>
+        /// <param name="type">Reference to the calling object.</param>
         /// <param name="alternateKey">Dictionary of alias and structural properties for the alternate key.</param>
-        public static void AddAlternateKey(this IEdmEntityType type, EdmModel model, IDictionary<string, IEdmProperty> alternateKey)
+        public static void AddAlternateKeyAnnotation(this EdmModel model, IEdmEntityType type, IDictionary<string, IEdmProperty> alternateKey)

```

Comment:
This was a design that I considered, adding the extension on Model instead of IEdmEntityType. Since you are adding the extension on EdmModel and not the interface, would it be better to add it in the actual EdmModel class itself?